CMakeLists.txt

```diff
@@ -16,7 +16,44 @@ if (WIN32)
     set(CMAKE_STATIC_LINKER_FLAGS "${CMAKE_STATIC_LINKER_FLAGS} /WX")
 endif()
 add_subdirectory(third_party/pybind11)
-FIND_PACKAGE(Protobuf REQUIRED)
+
+# Customized version of find Protobuf. We need to avoid situations mentioned
+# in https://github.com/caffe2/caffe2/blob/b7d983f255ef5496474f1ea188edb5e0ac442761/cmake/ProtoBuf.cmake#L82-L92
+# The following section is stolen from cmake/ProtoBuf.cmake in Caffe2
+find_program(PROTOBUF_PROTOC_EXECUTABLE
```

Comment:
Is there a way in find_program to set this as required?

Reply:
There is not. But if it fails to find `PROTOBUF_PROTOC_EXECUTABLE`, `PROTOBUF_PROTOC_EXECUTABLE` will be set to `PROTOBUF_PROTOC_EXECUTABLE-NOFOUND` and the downstream operation will fail with pretty clear message.